src/main/java/io/getunleash/UnleashClientFactory.java

```diff
@@ -0,0 +1,40 @@
+package io.getunleash;
+
+import io.getunleash.repository.FeatureRepository;
+import io.getunleash.strategy.Strategy;
+import io.getunleash.util.UnleashConfig;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UnleashClientFactory {
+
+    private static UnleashClientFactory INSTANCE;
+
+    private static Map<String, DefaultUnleash> unleashClients = new HashMap<>();
+
+    private UnleashClientFactory() {}
+
+    public static synchronized UnleashClientFactory getInstance() {
+        if (INSTANCE == null) {
```

Comment:
Is there a reason creating a instance of factory instead of using static methods and never initiating?